googledirectory/table_googledirectory_role_assignment.go

```diff
@@ -0,0 +1,130 @@
+package googledirectory
+
+import (
+	"context"
+
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+
+	admin "google.golang.org/api/admin/directory/v1"
+)
+
+//// TABLE DEFINITION
+
+func tableGoogleDirectoryRoleAssignment(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "googledirectory_role_assignment",
+		Description: "Role Assignments defined in the Google Workspace directory.",
+		List: &plugin.ListConfig{
+			Hydrate: listDirectoryRoleAssignments,
+			KeyColumns: []*plugin.KeyColumn{
+				{
+					Name:    "customer_id",
+					Require: plugin.Optional,
+				},
+			},
+		},
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("role_assignment_id"),
+			Hydrate:    getDirectoryRoleAssignment,
+		},
+		Columns: []*plugin.Column{
+			{
+				Name:        "role_assignment_id",
+				Description: "The unique ID for the role assigment.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "role_id",
+				Description: "The unique ID for the role.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "assigned_to",
+				Description: "The unique ID of the user this role is assigned to.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "customer_id",
+				Description: "The customer ID to retrieve all account roles.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromQual("customer_id"),
+			},
+			{
+				Name:        "etag",
+				Description: "A hash of the metadata, used to ensure there were no concurrent modifications to the resource when attempting an update.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "kind",
+				Description: "The type of the API resource.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "org_unit_id",
+				Description: "If the role is restricted to an organization unit, this contains the ID for the organization unit the exercise of this role is restricted to.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "scope_type",
+				Description: "The scope in which this role is assigned.",
+				Type:        proto.ColumnType_STRING,
+			},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listDirectoryRoleAssignments(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Create service
+	service, err := AdminService(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+
+	// Set default value to my_customer, to represent current account
+	customerID := "my_customer"
+	if d.KeyColumnQuals["customer_id"] != nil {
+		customerID = d.KeyColumnQuals["customer_id"].GetStringValue()
+	}
+
+	resp := service.RoleAssignments.List(customerID)
```

Comment:
Add support to filter using `roleId` and `userKey` also.
https://pkg.go.dev/google.golang.org/api@v0.52.0/admin/directory/v1#RoleAssignmentsListCall.RoleId
https://pkg.go.dev/google.golang.org/api@v0.52.0/admin/directory/v1#RoleAssignmentsListCall.UserKey